spyder/plugins/completion/kite/plugin.py

```diff
@@ -14,37 +14,73 @@
 from qtpy.QtCore import Slot
 
 # Local imports
+from spyder.config.base import running_under_pytest
 from spyder.config.manager import CONF
 from spyder.utils.programs import run_program
 from spyder.api.completion import SpyderCompletionPlugin
 from spyder.plugins.completion.kite.client import KiteClient
 from spyder.plugins.completion.kite.utils.status import (
     check_if_kite_running, check_if_kite_installed)
+from spyder.plugins.completion.kite.utils.install import (
+    KiteInstallationThread, FINISHED)
+from spyder.plugins.completion.kite.widgets.install import KiteInstallerDialog
 
 
 logger = logging.getLogger(__name__)
 
 
 class KiteCompletionPlugin(SpyderCompletionPlugin):
     COMPLETION_CLIENT_NAME = 'kite'
+    CONF_SECTION = 'kite'
+    CONF_FILE = False
 
     def __init__(self, parent):
         SpyderCompletionPlugin.__init__(self, parent)
         self.available_languages = []
         enable_code_snippets = CONF.get('lsp-server', 'code_snippets')
         self.client = KiteClient(None, enable_code_snippets)
         self.kite_process = None
+        self.kite_installation_thread = KiteInstallationThread(self)
+        # TODO: Connect thread status to status bar
+        self.kite_installer = KiteInstallerDialog(
+            parent,
+            self.kite_installation_thread)
         self.client.sig_client_started.connect(self.http_client_ready)
         self.client.sig_response_ready.connect(
             functools.partial(self.sig_response_ready.emit,
                               self.COMPLETION_CLIENT_NAME))
+        self.kite_installation_thread.sig_installation_status.connect(
+            lambda status: self.client.start() if status == FINISHED else None)
+        self.main.sig_setup_finished.connect(self.mainwindow_setup_finished)
 
     @Slot(list)
     def http_client_ready(self, languages):
         logger.debug('Kite client is available for {0}'.format(languages))
         self.available_languages = languages
         self.sig_plugin_ready.emit(self.COMPLETION_CLIENT_NAME)
 
+    @Slot()
+    def mainwindow_setup_finished(self):
+        """
+        Called when the setup of the main window finished
+        to let us do onboarding if necessary
+        :return:
+        """
+        self._show_installation_dialog()
+
+    def _show_installation_dialog(self):
+        """Show installation dialog."""
+        kite_installation_enabled = self.get_option('installation/enabled')
+        installed, path = check_if_kite_installed()
+        if (not installed and kite_installation_enabled
+                and not running_under_pytest()):
+            self.kite_installer.show()
+            self.kite_installer.center()
+
+    def get_option(self, option):
```

Comment:
Was not this method part of the parent class `SpyderCompletionPlugin` ?

Reply:
Yep, is part of BasePlugin/SpyderPlugin. I will remove this from here.

Reply:
Also, should we delete the `get_option` method over the LSP plugin [here](https://github.com/spyder-ide/spyder/blob/13c208033a144fe5bf2cb1f8f085c2d9b7cb0faa/spyder/plugins/completion/languageserver/plugin.py#L72)?

Reply:
I think so, could you test?

Reply:
Yes, now that method is no longer needed.